allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -29,27 +31,52 @@ class PretrainedTransformerIndexer(TokenIndexer):
         We use a somewhat confusing default value of ``tags`` so that we do not add padding or UNK
         tokens to this namespace, which would break on loading because we wouldn't find our default
         OOV token.
+    intra_word_tokenization: ``bool``, optional (default=False)
+        See superclass.
+    num_added_start_tokens: ``int``, optional (default=1)
+        The number of start tokens that the tokenizer adds to a sequence.
+    num_added_end_tokens: ``int``, optional (default=1)
+        The number of end tokens that the tokenizer adds to a sequence.
+    intra_word_tokenization: ``bool``, optional (default=False)
+        If True, further tokenize each token into subword tokens using the corresponding tokenizer.
+        Should be set to the same value as the ``intra_word_tokenized`` option on the
+        :class:`PretrainedTransformerEmbedder`.
     """
 
     def __init__(
-        self, model_name: str, namespace: str = "tags", token_min_padding_length: int = 0
+        self,
+        model_name: str,
+        namespace: str = "tags",
+        token_min_padding_length: int = 0,
+        num_added_start_tokens: int = 1,
+        num_added_end_tokens: int = 1,
+        intra_word_tokenization: bool = False,
     ) -> None:
         super().__init__(token_min_padding_length)
         self._namespace = namespace
-        self._tokenizer = AutoTokenizer.from_pretrained(model_name)
-        self._padding_value = self._tokenizer.convert_tokens_to_ids([self._tokenizer.pad_token])[0]
-        logger.info(f"Using token indexer padding value of {self._padding_value}")
+        # add_special_tokens=False for intra_word_tokenization
+        self._tokenizer = PretrainedTransformerTokenizer(model_name, add_special_tokens=False)
```

Comment:
Why do you need our tokenizer, instead of just the huggingface one?  What does the allennlp one get you?

Reply:
The indexer takes `Token`s as input, but (all) tokenizers take raw text. When doing subword tokenization I have to extract raw text from the tokens, tokenize, and then convert back to `Token`s. This last step is what our tokenizer is for. See `_intra_word_tokenize()` at the bottom of the file.

Reply:
Ok, yeah, that makes sense.